frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/storage-tab-state-update.ts

```diff
@@ -38,7 +51,17 @@ export const prefillInitialDiskUpdater = ({ id, prevState, dispatch, getState }:
       new DataVolumeWrapper(newSourceStorage.dataVolume).getType(),
     );
 
-  if (newType !== oldType) {
+  const baseDiskImageChanged = hasVmSettingsChanged(
+    prevState,
+    state,
+    id,
+    VMSettingsField.OPERATING_SYSTEM,
+    VMSettingsField.FLAVOR,
+    VMSettingsField.WORKLOAD_PROFILE,
+    VMSettingsField.CLONE_COMMON_BASE_DISK_IMAGE,
+  );
+
+  if (newType !== oldType || baseDiskImageChanged) {
```

Comment:
my biggest concern here is that we might be rewriting size/sc/url of already modified disks by the user. This can happen for all provision sources since it checks for changes in so many fields.

IMO we should check just for changes of pvc name and pvc namespace to detect changes between oldSourceStorage and newSourceStorage.
This way user changes should persist.

Reply:
this code preside this PR, changing this behaviour may cause problems  we can't see right now, we should look at it more carefully outside the scope of this PR

Reply:
this obviously breaks the current behaviour. What is wrong with the approach I just described?

Reply:
done